packages/atomic/src/components/facets-v1/facet-number-input/facet-number-input.tsx

```diff
@@ -0,0 +1,114 @@
+import {Component, h, State, Prop, Host} from '@stencil/core';
+import {
+  NumericFilter,
+  NumericFilterState,
+  loadNumericFacetSetActions,
+} from '@coveo/headless';
+import {Bindings} from '../../../utils/initialization-utils';
+
+/**
+ * Internal component made to be integrated in a NumericFacet.
+ * @internal
+ */
+@Component({
+  tag: 'atomic-facet-number-input',
+  shadow: false,
+})
+export class FacetNumberInput {
+  @State() private isRangeInvalid = false;
+  @State() private start?: number;
+  @State() private end?: number;
+
+  @Prop() public bindings!: Bindings;
+  @Prop() public facetId!: string;
+  @Prop() public filter!: NumericFilter;
+  @Prop() public filterState!: NumericFilterState;
+  @Prop() public label!: string;
+
+  connectedCallback() {
+    this.start = this.filterState.range?.start;
+    this.end = this.filterState.range?.end;
+  }
+
+  private onApply() {
+    this.bindings.engine.dispatch(
+      loadNumericFacetSetActions(
+        this.bindings.engine
+      ).deselectAllNumericFacetValues(this.facetId)
+    );
+
+    try {
+      this.filter.setRange({
+        start: this.start!,
+        end: this.end!,
+      });
+      this.isRangeInvalid = false;
+    } catch (error) {
+      this.bindings.engine.logger.error(error);
+      this.isRangeInvalid = true;
+    }
+  }
+
+  private get applyEnabled() {
+    return (
+      this.start !== undefined &&
+      this.end !== undefined &&
+      !isNaN(this.start) &&
+      !isNaN(this.end) &&
+      this.start >= 0 &&
```

Comment:
There *could* be scenarios where having a negative start value would be okay 🤔 
So i'm not sure that it's a valid check here.